api/v1alpha1/dockermachine_types.go

```diff
@@ -22,6 +22,8 @@ import (
 
 // DockerMachineSpec defines the desired state of DockerMachine
 type DockerMachineSpec struct {
+	// ProviderID will be the container name in ProviderID format (docker:////<containername>)
```

Comment:
I believe we also need `// +optional` (although the CAPA field doesn't currently have it) - @detiber?

Reply:
ah, good catch, yes should likely be optional.

Reply:
oh yes, totally thanks

Reply:
hmm that didn't change any generated code...🤔

Reply:
It affects openapi

Reply:
oh that must be just for validation, ok, i think that makes sense then

Reply:
It should generate changes to the crd yaml

Reply:
`make generate`, which does indeed run generate across ./api/... does not modify the crd yaml with this change.

Reply:
Looking closer at the generated CRD, since it doesn't (and didn't) specify required on the object, maybe kubebuilder automatically treats pointers w/ omitempty as optional?

Reply:
old version of controller-gen. My fault.